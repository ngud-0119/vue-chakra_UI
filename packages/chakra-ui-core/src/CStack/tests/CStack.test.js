import { CStack, CBox, CHeading, CText } from '../..'
import { render, getTagName } from '@/tests/test-utils'

const renderComponent = (props) => {
  const inlineAttrs = (props && props.inlineAttrs) || ''
  const base = {
    components: { CStack, CBox, CHeading, CText },
    template: `
    <CStack data-testid="stack" ${inlineAttrs}>
      <CBox p="5" data-testid="stack-child-1" shadow="md" border-width="1px">
        <CHeading font-size="xl">Plan Money</CHeading>
        <CText mt="4">The future can be even brighter but a goal without a plan is just a wish</CText>
      </CBox>
      <CBox p="5" data-testid="stack-child-2" shadow="md" border-width="1px">
        <CHeading font-size="xl">Save Money</CHeading>
        <CText mt="4">You deserve good things. With a whooping 10-15% interest rate per annum, grow your savings on your own terms with our completely automated process</CText>
      </CBox>
    </CStack>`,
    ...props
  }
  return render(base)
}

it('should render correctly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should default to vertical stack', () => {
  const { getByTestId } = renderComponent()
  const stack = getByTestId('stack')
  expect(stack).toHaveStyle('display: flex')
  expect(stack).toHaveStyle('flex-direction: column')
})

it('should not space last child', () => {
  const { getByTestId } = renderComponent()
  const stack = getByTestId('stack')
  expect(stack).not.toHaveStyle('margin-bottom: 0.5rem')
})

it('should should stack horizontally if isInline', () => {
  const inlineAttrs = 'is-inline'
  const { getByTestId } = renderComponent({ inlineAttrs })
  const stack = getByTestId('stack')
  expect(stack).toHaveStyle('display: flex')
  expect(stack).toHaveStyle('flex-direction: row')
})

// Cannot use `it.each` because it cannot accept
// component as interpolated variable

it.each`
  as
  ${'section'}
  ${'nav'}}
`(
  'should render CStack with element as $as',
  ({ as }) => {
    const inlineAttrs = `as="${as}"`
    const { asFragment, getByTestId } = renderComponent({ inlineAttrs })
    const stack = getByTestId('stack')
    expect(getTagName(stack)).toEqual(as)
    expect(asFragment()).toMatchSnapshot()
  }
)
