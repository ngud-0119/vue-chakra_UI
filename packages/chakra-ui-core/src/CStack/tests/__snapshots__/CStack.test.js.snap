// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CStack should render CStack with element as nav 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  padding: var(--chakra-space-5);
  box-shadow: var(--chakra-shadows-md);
  border-width: 1px;
  margin-bottom: var(--chakra-space-2);
}

.emotion-3 {
  font-size: var(--chakra-fontSizes-xl);
  line-height: var(--chakra-lineHeights-shorter);
  font-weight: var(--chakra-fontWeights-bold);
  font-family: var(--chakra-fonts-heading);
}

.emotion-4 {
  font-family: var(--chakra-fonts-body);
  margin-top: var(--chakra-space-4);
}

.emotion-5 {
  padding: var(--chakra-space-5);
  box-shadow: var(--chakra-shadows-md);
  border-width: 1px;
}

<nav
    class="emotion-0 emotion-1"
    data-chakra-component="CStack"
    data-testid="stack"
  >
    <div
      class="emotion-2"
      data-chakra-component="CBox"
      data-testid="stack-child-1"
    >
      <h2
        class="emotion-3"
        data-chakra-component="CHeading"
      >
        Plan Money
      </h2>
       
      <p
        class="emotion-4"
        data-chakra-component="CText"
      >
        The future can be even brighter but a goal without a plan is just a wish
      </p>
    </div>
    <div
      class="emotion-5"
      data-chakra-component="CBox"
      data-testid="stack-child-2"
    >
      <h2
        class="emotion-3"
        data-chakra-component="CHeading"
      >
        Save Money
      </h2>
       
      <p
        class="emotion-4"
        data-chakra-component="CText"
      >
        You deserve good things. With a whooping 10-15% interest rate per annum, grow your savings on your own terms with our completely automated process
      </p>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`CStack should render CStack with element as section 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  padding: var(--chakra-space-5);
  box-shadow: var(--chakra-shadows-md);
  border-width: 1px;
  margin-bottom: var(--chakra-space-2);
}

.emotion-3 {
  font-size: var(--chakra-fontSizes-xl);
  line-height: var(--chakra-lineHeights-shorter);
  font-weight: var(--chakra-fontWeights-bold);
  font-family: var(--chakra-fonts-heading);
}

.emotion-4 {
  font-family: var(--chakra-fonts-body);
  margin-top: var(--chakra-space-4);
}

.emotion-5 {
  padding: var(--chakra-space-5);
  box-shadow: var(--chakra-shadows-md);
  border-width: 1px;
}

<section
    class="emotion-0 emotion-1"
    data-chakra-component="CStack"
    data-testid="stack"
  >
    <div
      class="emotion-2"
      data-chakra-component="CBox"
      data-testid="stack-child-1"
    >
      <h2
        class="emotion-3"
        data-chakra-component="CHeading"
      >
        Plan Money
      </h2>
       
      <p
        class="emotion-4"
        data-chakra-component="CText"
      >
        The future can be even brighter but a goal without a plan is just a wish
      </p>
    </div>
    <div
      class="emotion-5"
      data-chakra-component="CBox"
      data-testid="stack-child-2"
    >
      <h2
        class="emotion-3"
        data-chakra-component="CHeading"
      >
        Save Money
      </h2>
       
      <p
        class="emotion-4"
        data-chakra-component="CText"
      >
        You deserve good things. With a whooping 10-15% interest rate per annum, grow your savings on your own terms with our completely automated process
      </p>
    </div>
  </section>
</DocumentFragment>
`;

exports[`CStack should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  padding: var(--chakra-space-5);
  box-shadow: var(--chakra-shadows-md);
  border-width: 1px;
  margin-bottom: var(--chakra-space-2);
}

.emotion-3 {
  font-size: var(--chakra-fontSizes-xl);
  line-height: var(--chakra-lineHeights-shorter);
  font-weight: var(--chakra-fontWeights-bold);
  font-family: var(--chakra-fonts-heading);
}

.emotion-4 {
  font-family: var(--chakra-fonts-body);
  margin-top: var(--chakra-space-4);
}

.emotion-5 {
  padding: var(--chakra-space-5);
  box-shadow: var(--chakra-shadows-md);
  border-width: 1px;
}

<div
    class="emotion-0 emotion-1"
    data-chakra-component="CStack"
    data-testid="stack"
  >
    <div
      class="emotion-2"
      data-chakra-component="CBox"
      data-testid="stack-child-1"
    >
      <h2
        class="emotion-3"
        data-chakra-component="CHeading"
      >
        Plan Money
      </h2>
       
      <p
        class="emotion-4"
        data-chakra-component="CText"
      >
        The future can be even brighter but a goal without a plan is just a wish
      </p>
    </div>
    <div
      class="emotion-5"
      data-chakra-component="CBox"
      data-testid="stack-child-2"
    >
      <h2
        class="emotion-3"
        data-chakra-component="CHeading"
      >
        Save Money
      </h2>
       
      <p
        class="emotion-4"
        data-chakra-component="CText"
      >
        You deserve good things. With a whooping 10-15% interest rate per annum, grow your savings on your own terms with our completely automated process
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CStack should render empty children correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
    class="emotion-0 emotion-1"
    data-chakra-component="CStack"
  />
</DocumentFragment>
`;

exports[`CStack should should stack native html elements 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  font-family: var(--chakra-fonts-body);
  margin-top: var(--chakra-space-4);
  margin-bottom: var(--chakra-space-2);
}

.emotion-3 {
  margin-bottom: var(--chakra-space-2);
}

.emotion-5 {
  font-family: var(--chakra-fonts-body);
  margin-top: var(--chakra-space-4);
}

<div
    class="emotion-0 emotion-1"
    data-chakra-component="CStack"
    data-testid="stack"
  >
    <p
      class="emotion-2"
      data-chakra-component="CText"
    >
      The future can be even brighter but a goal without a plan is just a wish
    </p>
    <div
      class="emotion-3"
      data-chakra-component="CBox"
    >
      <p
        data-testid="stacked-p"
      >
        I am a happy paragraph element
      </p>
    </div>
    <div
      class="emotion-3"
      data-chakra-component="CBox"
    >
      <h3
        data-testid="stacked-h3"
      >
        I am a jolly heading element
      </h3>
    </div>
    <p
      class="emotion-5"
      data-chakra-component="CText"
    >
      You deserve good things. With a whooping 10-15% interest rate per annum, grow your savings on your own terms with our completely automated process
    </p>
  </div>
</DocumentFragment>
`;
